### DISCUSSION TITLE
SwitchMode1 11 on SONOFF D1 triggers all the time state=4 no matter if button pressed or not

### DISCUSSION THREAD

**Discussion by samezrp** [arendst/Tasmota#16332] [Q&A] [answered]
**Tasmota: 12.1.0
HW: SONOFF D1
GPIO0: set to Switch 1 (checked with switch_n as well)
Setup: Backlog SwitchMode1 11; SetOption32 10;
**
Switch states tested with:

`rule1 on switch1#state do publish switch_test %value% endon`

On SONOFF D1 there is a bug (I believe) and switch _state=4_ under SwitchMode1 11 is triggered all the time, no matter if switch is pressed or not.

One press gives:
```
**endless**
18:19:06.772 RSL: switch_test = 4
18:19:07.170 RUL: SWITCH1#STATE performs "publish switch_test 4"
18:19:07.174 RSL: switch_test = 4
18:19:07.644 RUL: SWITCH1#STATE performs "publish switch_test 4"
18:19:07.648 RSL: switch_test = 4
18:19:07.916 RUL: SWITCH1#STATE performs "publish switch_test 7"
18:19:07.919 RSL: switch_test = 7
18:19:08.114 RUL: SWITCH1#STATE performs "publish switch_test 5"
18:19:08.117 RSL: switch_test = 5
18:19:08.717 RUL: SWITCH1#STATE performs "publish switch_test 4"
**endless**
```
long press gives:
```
**endless**
18:21:07.274 RSL: switch_test = 4
18:21:07.670 RUL: SWITCH1#STATE performs "publish switch_test 4"
18:21:07.674 RSL: switch_test = 4
18:21:08.072 RUL: SWITCH1#STATE performs "publish switch_test 4"
18:21:08.076 RSL: switch_test = 4
18:21:08.466 RUL: SWITCH1#STATE performs "publish switch_test 6"
18:21:08.470 RSL: switch_test = 6
18:21:08.484 RUL: SWITCH1#STATE performs "publish switch_test 7"
18:21:08.488 RSL: switch_test = 7
18:21:09.470 RUL: SWITCH1#STATE performs "publish switch_test 6"
18:21:09.474 RSL: switch_test = 6
**stop until button is pressed**
18:21:16.022 RUL: SWITCH1#STATE performs "publish switch_test 4"
18:21:16.026 RSL: switch_test = 4
18:21:16.418 RUL: SWITCH1#STATE performs "publish switch_test 4"
18:21:16.422 RSL: switch_test = 4
18:21:16.817 RUL: SWITCH1#STATE performs "publish switch_test 4"
18:21:16.821 RSL: switch_test = 4
**endless**
```
so there is no way to have this SwitchMode1 11 working with endless switch_test = 4 as described in manual:
https://tasmota.github.io/docs/Buttons-and-Switches/#switchmode

SwitchMode1 12 could be an option as there is no this endless state = 4 action, however it does not create switch status = 5, so there is no way to change dim direction. It could be done via status=10, however it is not natural and would give another step when dimmed.

It looks like:
18:01:51.130 RSL: RESULT = {"SwitchMode1":12}

press

```
18:03:15.114 RUL: SWITCH1#STATE performs "publish switch_test 2"
18:03:15.118 RSL: switch_test = 2
18:03:15.714 RUL: SWITCH1#STATE performs "publish switch_test 10"
18:03:15.718 RSL: switch_test = 10
18:03:15.731 RUL: SWITCH1#STATE performs "publish switch_test 6"
18:03:15.735 RSL: switch_test = 6

```

long press

```
18:03:56.032 RSL: switch_test = 4
18:03:56.428 RUL: SWITCH1#STATE performs "publish switch_test 4"
18:03:56.432 RSL: switch_test = 4
18:03:56.827 RUL: SWITCH1#STATE performs "publish switch_test 4"
18:03:56.831 RSL: switch_test = 4
18:03:57.227 RUL: SWITCH1#STATE performs "publish switch_test 4"
18:03:57.231 RSL: switch_test = 4
18:03:57.698 RUL: SWITCH1#STATE performs "publish switch_test 4"
18:03:57.702 RSL: switch_test = 4
18:03:58.098 RUL: SWITCH1#STATE performs "publish switch_test 4"
18:03:58.102 RSL: switch_test = 4
**state = 4 repeated until button pressed, then**
18:03:58.199 RUL: SWITCH1#STATE performs "publish switch_test 7"
18:03:58.203 RSL: switch_test = 7
18:03:59.198 RUL: SWITCH1#STATE performs "publish switch_test 6"
18:03:59.202 RSL: switch_test = 6
```

**Comment by sfromis** [comment, marked as answer]
If SwitchMode 11 is registering as pushed all the time (which your logs strongly suggests), that indeed means that SwitchMode 12 is not just "an option", but specifically required.

**Comment by samezrp** [comment]
If it would be as you describe that is OK, however in SwitchMode 12 nor 11 there is no state = 5 triggered, and as I understand is needed to invert dimming after long press.
Another thing is that in this case it is not working as described in manual:

> **When pressing the button (closing the circuit) for a long time** (set in SetOption32), Tasmota will send repeated INC_DEC (increment or decrement the dimmer) commands for **as long as the button is pressed** (**use Switch<x>#state=4 in rules**).

**Reply by barbudor** [reply]
> If the button is pressed again before the timeout, Tasmota will send an INV command. The INV command is for the controlling software (Home Assistant) to switch between incrementing and decrementing the dimmer (use Switch<x>#state=5 in rules).

Release the button and press again before the end of the timer and you will have state 5

**Reply by samezrp** [reply]
You are right, when double press there is:

```
22:35:27.106 RUL: SWITCH1#STATE performs "publish switch_test 2"
22:35:27.110 RSL: switch_test = 2
22:35:27.214 RUL: SWITCH1#STATE performs "publish switch_test 5"
22:35:27.218 RSL: switch_test = 5
22:35:27.362 RUL: SWITCH1#STATE performs "publish switch_test 8"
22:35:27.366 RSL: switch_test = 8
22:35:28.366 RUL: SWITCH1#STATE performs "publish switch_test 6"
22:35:28.370 RSL: switch_test = 6

```
But it makes no sense and still: it is not according to manual.
In fact what you wrote means that I have to double click to reverse dimming.
I still claim state = 4 status triggered when switch is not pressed as behavior not according to manual.
